connection.js

```diff
@@ -271,25 +272,54 @@ class Connection {
             if (config.DEBUG) console.log("There are no connected games");
         }
 
-        if (config.maxtotalgames && this.connected_games && Object.keys(this.connected_games).length >= config.maxtotalgames){
-            conn_log(Object.keys(this.connected_games).length + " games being played, maximum is " + config.maxtotalgames);
-            return { reject: true, msg: "Currently, " + Object.keys(this.connected_games).length + " games are being played by this bot, maximum is " + config.maxtotalgames + " (if you see this message and you dont see any game on the bot profile page, it is because private game(s) are being played) , try again later " };
+        if (config.maxconnectedgames && this.connected_games && Object.keys(this.connected_games).length >= config.maxconnectedgames){
+            conn_log(Object.keys(this.connected_games).length + " games being played, maximum is " + config.maxconnectedgames);
+            return { reject: true, msg: "Currently, " + Object.keys(this.connected_games).length + " games are being played by this bot, maximum is " + config.maxconnectedgames + " (if you see this message and you dont see any game on the bot profile page, it is because private game(s) are being played) , try again later " };
         }
 
-        if (user.ranking < config.minrank) {
+        if ((user.ranking < config.minrank) && !config.minrankranked && !config.minrankunranked) {
             let humanReadableUserRank = rankToString(user.ranking);
             let humanReadableMinRank = rankToString(config.minrank);
             conn_log(user.username + " ranking too low: " + humanReadableUserRank + " : min is " + humanReadableMinRank);
             return { reject: true, msg: "Minimum rank is " + humanReadableMinRank + ", your rank is too low." };
         }
 
-        if (user.ranking > config.maxrank) {
+        if ((user.ranking < config.minrankranked) && notification.ranked) {
+            let humanReadableUserRank = rankToString(user.ranking);
+            let humanReadableMinRank = rankToString(config.minrankranked);
+            conn_log(user.username + " ranking too low: " + humanReadableUserRank + " : min for ranked games is " + humanReadableMinRank);
+            return { reject: true, msg: "Minimum rank for ranked games is " + humanReadableMinRank + ", your rank is too low, try unranked game" };
```

Comment:
The only difference with line 280-284 is the return value here. I got the feeling this could be done with a lot less duplicate code.

The check on line 280 could be `config.minrank && user.ranking < config.minrank || user.ranking && user.ranking < config.minrankranked` and then you only need to return a different message based on `config.minrank` like

```
`${user.username} ranking too low: ${humanReadableUserRank}: min ${config.minrank ? "" : for ranked games} is ${humanReadableMinRank}`
```

btw, consider using string interpolation everywhere.

Ps. I removed another redundant space here. Please keep it clean and remove all spaces before interpunction like `:`, `,`, `.` etc. etc.

Reply:
this is quite complicated @Dorus :rofl:

Reply:
oh i see you mean to use `blablah ${1} blablah` instead of "blablah " + v1 + " blablah";

i see but i did it to keep in sync with the current code
but also because on gedit the "blablah" is pink so easy to read, but the `blablah` is dark same color as plain text, hard to read

Reply:
also if i start changing this then i would have to change all the other ones for consistency :sob: 

but yes it's better i think ! ty for information !

Reply:
Ah yes i got the same color problem in my editor. Really need to look into a way to fix that first i think :(